### PR TITLE
[Flutter] device ID interface documentation

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Review by turtledreams — CHANGES_REQUESTED** [review]
lets not advertise merge/nonmerge calls
lets only provide setID
and we have to keep the old call still because 7.0 users will have to use that. find a creative way to display both pls

**Review by turtledreams — CHANGES_REQUESTED** [review]
hmm did not like this. Lets make like this: discard this commit and in previous commit add a link to the previous doc's device id section in the callout and mention 7.0 should use that method.

**Review by turtledreams — CHANGES_REQUESTED** [review]
current is 24.7.0. you should not create a new file for 24.7.0. You should link to an older documentation

**Review by turtledreams — CHANGES_REQUESTED** [review]
1. links in next and current are different
2. now you removed mentioning setID and temp calls are available from 24.7.1 => customer dont know why they should use those links

File: flutter/current.md [hunk 1]
```diff
@@ -684,77 +684,79 @@ config.setGlobalViewSegmentation(segmentation);</code></pre>
...
-  in mind that it has to be unique for all users. Some potential sources for such
-  an id could be the username, email or some other internal ID used by your other
-  systems.
+  ID or allow the SDK to generate it. When providing your device ID, ensure it
+  is unique for all users. Potential sources for such an ID include the username,
+  email, or other internal ID used by your other systems.
 </p>
+<div class="callout callout--info">
+  <p>
+    <strong>Click <a href="https://support.countly.com/hc/en-us/articles/34539364044697" target="_blank" rel="noopener noreferrer">here</a>  to access the documentation for version 24.7.0</strong>
```
Comment (arifBurakDemiray) [inline flutter/current.md:693]: just to simplify this link, we can use hc/en-us/articles/34539364044697 to make it more pleasant I think

@turtledreams what do you think? Should we all go for short version

**Review by arifBurakDemiray — COMMENTED** [review]
🤖

File: flutter/current.md (hunk 1 above)
Comment (turtledreams) [reply flutter/current.md:693]: when viewed it is nice, true, but I dont want to micromanage to that level, as already the iteration on bigger things taking time. when a link is added through zendesk's ui it should be shortening it as you mentioned, if possible I prefer that too, but not mandatory

**Review by turtledreams — APPROVED** [review]

**Pull request merged**